.github/workflows/main.yml

```diff
@@ -17,24 +17,28 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+
       - name: Set up Python 3.8
         uses: actions/setup-python@v1
         with:
           python-version: 3.8
+
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install --upgrade pipenv
-          pipenv install --dev --system --ignore-pipfile
-      - name: Lint with flake8
-        run: |
-          flake8
-      - name: Lint with black
-        run: |
-          black --check .
-      - name: Lint with mypy
-        run: |
-          mypy --check ggshield
+          python -m pip install pipenv pre-commit
+          pipenv install --dev --skip-lock
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
```

Comment:
Otherwise looks good to me.

Reply:
I copied this from https://github.com/pre-commit/action/blob/main/action.yml (I didn't use the action this code comes from because the [README](https://github.com/pre-commit/action) says it's deprecated).

I understand this ensures the cache is not reused if the Python version changes, because pythonLocation includes the Python version.